File: typescript/base-hmi-types/src/screens/base/HmiFaceplateContainer.ts
import { HmiContainerBase } from "./HmiContainerBase.js";
import { HmiObjectType } from "./HmiObjectType.js";
import { HmiFaceplateInterfaceValue } from "./HmiFaceplateInterfaceValue.js";

export class HmiFaceplateContainer extends HmiContainerBase {
  faceplateId?: string;
  faceplateName?: string;
  faceplateVersion?: string;
  readonly interfaceValues: HmiFaceplateInterfaceValue[] = [];

  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiFaceplateContainer;
  }
}

File: src/BaseHmiTypes/Screens/Base/HmiFaceplateContainer.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiFaceplateContainer : HmiContainerBase
{
    public HmiFaceplateContainer()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiFaceplateContainer;
    }

    public string? FaceplateId { get; set; }

    public string? FaceplateName { get; set; }

    public string? FaceplateVersion { get; set; }

    public IList<HmiFaceplateInterfaceValue> InterfaceValues { get; } = new List<HmiFaceplateInterfaceValue>();
}
